expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,837 @@
+        """Plot a line showing the threshold.
+
+        Parameters
+        ----------
+            n_skip : int
+                See documentation for ``TrackerUD.threshold``.
+            ax : Axes
+                The handle to the axes object. If None, the current axes will
+                be used.
+
+        Returns
+        -------
+        line : list Line2D
+            The handle to the threshold line, as returned from ``plt.plot``.
+        """
+        if ax is None:
+            ax = plt.gca()
+        h = ax.plot([1, self._n_trials], [self.threshold(n_skip)] * 2,
+                    '--', color='gray')
+        return h
+
+    def threshold(self, n_skip=2):
+        """Plot a line showing the threshold on the current axes.
+
+        Parameters
+        ----------
+        n_skip : int
+            The number of reversals to skip at the beginning when computing
+            the threshold.
+
+        Returns
+        -------
+        threshold : float
+            The handle to the threshold line.
+
+        Notes
+        -----
+        The threshold is computed as the average of the up reversals and the
+        average of the down reversals. In this way if there's an unequal number
+        of them the assymetry won't bias the threshold estimate.
+        """
+        rev_inds = self.reversal_inds[n_skip:]
+        if len(rev_inds) < 1:
+            return np.nan
+        else:
+            return (np.mean(self._x[rev_inds[0::2]]) +
+                    np.mean(self._x[rev_inds[1::2]])) / 2
+
+
+# =============================================================================
+# Define the TrackerBinom Class
+# =============================================================================
+class TrackerBinom(object):
+    """Binomial hypothesis testing tracker
+
+    This class implements a tracker that runs a test at each trial with the
+    null hypothesis that the stimulus presented has no effect on the subject's
+    response. This would happen in the case where the subject couldn't hear a
+    stimulus, they were not able to do a task, or they did not understand the
+    task. This function's main use is training: a subject may move on to the
+    experiment when the null hypothesis that they aren't doing the task has
+    been rejected.
+
+    Parameters
+    ----------
+    callback : callable | ExperimentController | None
+        The function that will be used to print the data, usually to the
+        experiment .tab file. It should follow the prototype of
+        ``ExperimentController.write_data_line``. If an instance of
+        ``ExperimentController`` is given, then it will take that object's
+        ``write_data_line`` function. If ``None`` is given, then it will not
+        write the data anywhere.
+    alpha : float
+        The p-value which is considered significant. Must be between 0 and 1.
+        Note that if ``stop_early`` is `true` there is the potential for
```

Comment:
Done (sorry about all the "dones" you are about to receive, I'm just trying to keep track)